cfgov/v1/tests/models/test_sublanding_page.py

```diff
@@ -0,0 +1,89 @@
+import mock
+
+from unittest import TestCase
+from v1.models import SublandingPage, BrowseFilterablePage, AbstractFilterPage
+from wagtail.wagtailcore.blocks import StreamValue
+
+import scripts._atomic_helpers as atomic
+from v1.tests.wagtail_pages import helpers
+
+import datetime as dt
+
+
+class SublandingPageTestCase(TestCase):
+    """
+    This test case checks that the browse-filterable posts of a sublanding page are
+    properly retrieved.
+    """
+    def setUp(self):
+        self.request = mock.MagicMock()
+        self.request.site.hostname = 'localhost:8000'
+        self.limit = 10
+        self.sublanding_page = SublandingPage(title='title')
+
+        helpers.publish_page(child=self.sublanding_page)
+        self.post1 = BrowseFilterablePage(title='post 1')
+        self.post2 = BrowseFilterablePage(title='post 2')
+        # the content of this post has both a full_width_text and a filter_controls
+        self.post1.content = StreamValue(self.post1.content.stream_block,
+                                         [atomic.full_width_text, atomic.filter_controls], True)
+        # this one only has a filter_controls
+        self.post2.content = StreamValue(self.post1.content.stream_block,
+                                         [atomic.filter_controls], True)
+
+        helpers.save_new_page(self.post1, self.sublanding_page)
+        helpers.save_new_page(self.post2, self.sublanding_page)
+
+        # manually set the publication date of the posts to ensure consistent
+        # order of retrieval in test situations, otherwise the `date_published`
+        # can vary due to commit order
```

Comment:
What I found was that when I ran the test several times in a row, I would get different results, i.e. occasionally the test would fail when previously it had succeeded (without anything being changed). When I looked at the order of the posts returned from get_browsefilterable_posts, I discovered that it would vary. It looks like the cause of this is that if you do not supply the date_published argument, it will be set to the current date, and when that happens the retrieval order will depend on, presumably, the order of commit. One way of making sure this doesn't become an issue for actual posts is to change the DateField that @higs4281 was referring to into a DateTimeField, which would save us from relying on commit order for posts that were published on the same date. For testing purposes, I just supplied a date manually so that the retrieval order would be consistent.